tests/dip_sip/test_dip_sip.py

```diff
@@ -0,0 +1,168 @@
+        pytest.fail("minigraph_portchannels is not defined")
+    host_facts = dut.setup()['ansible_facts']
+    # minigraph facts
+    src_lag = mg_facts['minigraph_portchannel_interfaces'][2]['attachto']
+    dst_lag = mg_facts['minigraph_portchannel_interfaces'][0]['attachto']
+    facts['src_port'] = src_lag
+    facts['dst_port'] = dst_lag
+    logger.info("src_lag is {}, dst_lag is {}".format(src_lag, dst_lag))
+    # lldp facts
+    lldp_facts = dut.lldp()['ansible_facts']['lldp']
+    facts['dst_host_mac'] = lldp_facts[mg_facts['minigraph_portchannels'][dst_lag]['members'][0]]['chassis']['mac']
+    facts['src_host_mac'] = lldp_facts[mg_facts['minigraph_portchannels'][src_lag]['members'][0]]['chassis']['mac']
+    facts['dst_router_mac'] = host_facts['ansible_' + dst_lag]['macaddress']
+    facts['src_router_mac'] = host_facts['ansible_' + src_lag]['macaddress']
+    facts['dst_router_ipv4'] = host_facts['ansible_' + dst_lag]['ipv4']['address']
+    dst_ipv6 = host_facts['ansible_' + dst_lag]['ipv6']
+    facts['dst_router_ipv6'] = [(item['address']) for item in dst_ipv6 if item['scope'] == 'global'][0]
+    facts['dst_port_ids'] = [mg_facts['minigraph_port_indices'][mg_facts['minigraph_portchannels'][dst_lag]['members'][0]]]
+    facts['src_port_ids'] = [mg_facts['minigraph_port_indices'][mg_facts['minigraph_portchannels'][src_lag]['members'][0]]]
+
+    return facts
+
+
+def port_facts(dut):
+    facts = {}
+    mg_facts = dut.minigraph_facts(host=dut.hostname)['ansible_facts']
+
+    if not mg_facts['minigraph_interfaces']:
+        pytest.fail("minigraph_interfaces is not defined.")
+    host_facts = dut.setup()['ansible_facts']
+    # minigraph facts
+    src_port = mg_facts['minigraph_interfaces'][2]['attachto']
+    dst_port = mg_facts['minigraph_interfaces'][0]['attachto']
+    facts['src_port'] = src_port
+    facts['dst_port'] = dst_port
+    logger.info("src_port is {}, dst_port is {}".format(src_port, dst_port))
+    # lldp facts
+    lldp_facts = dut.lldp()['ansible_facts']['lldp']
+    facts['dst_host_mac'] = lldp_facts[dst_port]['chassis']['mac']
+    facts['src_host_mac'] = lldp_facts[src_port]['chassis']['mac']
+    facts['dst_router_mac'] = host_facts['ansible_' + dst_port]['macaddress']
+    facts['src_router_mac'] = host_facts['ansible_' + src_port]['macaddress']
+    facts['dst_router_ipv4'] = host_facts['ansible_' + dst_port]['ipv4']['address']
+    dst_ipv6 = host_facts['ansible_' + dst_port]['ipv6']
+    facts['dst_router_ipv6'] = [(item['address']) for item in dst_ipv6 if item['scope'] == 'global'][0]
+    facts['dst_port_ids'] = [mg_facts['minigraph_port_indices'][dst_port]]
+    facts['src_port_ids'] = [mg_facts['minigraph_port_indices'][src_port]]
+
+    return facts
+
+
+@pytest.fixture(scope='function')
+def gather_facts(testbed_devices, testbed):
+    facts = {}
+    topo = testbed['topo']['name']
+    if topo not in TOPO_LIST:
+        pytest.skip("Unsupported topology")
```

Comment:
we need some uniform way to handle this skip. similiar pr. https://github.com/Azure/sonic-mgmt/pull/1468

@wangxin , @yxieca , comments?

Reply:
Agree. But how?

PR #1468 implemented a good way of getting topology type of each topology, for example:
ptf16, ptf32, ptf64, all of them are type `ptf`. 
All the topologies are treated as variations of 3 basic types: t0, t1, ptf. Using the `testbed` fixture can access the topology type. For example `testbed['topo']['type']`.

Now the next question is where to specify the topology types supported by each test script? For all the ansible tests, we specify them in file ansible\roles\test\vars\testcases.yml. But testcases.yml cannot be simply used for pytest so far. Because for pytest, we haven't define things like testcase name which is key for accessing information defined in testcases.yml.

Option A is to create a separate common file for storing supported topology of each test script. Information in the file can be indexed by filename of each test script. Then we can have a common fixture to do the supported topology check.  The disadvantage is that whenever we add a new script, we must remember to add supported topology type information in the common file.

Option B. I am not sure if this option is doable. The idea is to use some pytest hook function to enforce each test script has module variable `SUPPORTED_TOPOLOGY_TYPES` defined. If a test script does not have this module variable defined, then refuse to run it. Then we can use a fixture to check `testbed['topo']['type']` against value defined in this module variable.

Reply:
@wangxin _Option A_ looks good to me. The same approach is already used in Ansible.